datafusion/datasource-parquet/src/file_format.rs

```diff
@@ -959,14 +953,18 @@ pub async fn fetch_parquet_metadata(
     store: &dyn ObjectStore,
     meta: &ObjectMeta,
     size_hint: Option<usize>,
-    decryption_properties: Option<&FileDecryptionProperties>,
+    #[allow(unused)] decryption_properties: Option<&FileDecryptionProperties>,
```

Comment:
Would it be better to use `#[cfg(feature = "parquet_encryption")]` on this parameter so this isn't a breaking change unless the encryption feature is enabled? (assuming this PR goes in before the next release)

Reply:
Hmm yeah OK I agree. It would be nice to not need to change the API, but I guess this might be fine here as it's documented as "exposed for low level integrations".

Reply:
A nicer API over the long term would be to move the arguments of `fetch_parquet_metadata` to a struct, and then have a field in that struct be different if the feature was enabled. 

That could also be a different / follow on PR if we wanted to pursue it